app/models/user.rb

```diff
@@ -12,6 +16,8 @@ class User < ActiveRecord::Base
 
   has_secure_password validations: false
 
+  before_create :generate_token
```

Comment:
This is sleeping security threat. If you create token right after creation. Better option is to create it when requested. And have it expire after some short time.